### PR TITLE
CPU features for AMD

### PR DESCRIPTION
**Pull request by toor1245**
Hello everybody, In this pull requests I added improvements of cpu features of AMD microarchitecture related to https://github.com/google/cpu_features/issues/116

Results on real machine:

**ZEN2**
*  Before changes:
        image
*  After changes:
        image
As you can see, uarch changed  from **AMD_ZEN** to **AMD_ZEN2**

Fixed a test that checked the wrong microarchitecture, where instead of _AMD_BULLDOZER_ must be _AMD_STREAMROLLER_

Added processing of HygonGenuine vendor, since there is a microarchitecture ZEN in Family 24 (18h) Dhyana. [CPUID_Dump](http://users.atw.hu/instlatx64/HygonGenuine/HygonGenuine0900F02_Hygon_CPUID3.txt)

### CODE REVIEW

**Comment by toor1245** [comment]
@gchatelet @Mizux please review

File: src/cpuinfo_x86.c [hunk 1]
```diff
@@ -1518,26 +1519,119 @@ X86Microarchitecture GetX86Microarchitecture(const X86Info* info) {
...
+      case CPUID(0x19, 0x30):
+      case CPUID(0x19, 0x40):
+      case CPUID(0x19, 0x50):
+        // https://en.wikichip.org/wiki/amd/microarchitectures/zen_3
         return AMD_ZEN3;
       default:
         return X86_UNKNOWN;
     }
   }
+  if (memcmp(info->vendor, "HygonGenuine", sizeof(info->vendor)) == 0) {
```
Comment (gchatelet) [inline src/cpuinfo_x86.c:1629]: Could you reuse `IsVendor` here ?

**Comment by gchatelet** [comment]
Can you `clang-format -i include/cpuinfo_x86.h`?

Thx a lot for the PR!

**Comment by toor1245** [comment]
> Can you `clang-format -i include/cpuinfo_x86.h`?
> 
> Thx a lot for the PR!

done

**Pull request merged**